File: src/dotnet/yaxgl/YaxglLibrary/Controls/RadioButton.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace de.yaxgl
{
    public class RadioButton : Control
    {
        public RadioButton(Container owner, string ID)
        {
            this.owner = owner;
            this.ID = ID;
            this.control = new System.Windows.Forms.RadioButton();
            /*register events*/
            this.control.Click += new System.EventHandler(clickEvent);
            this.control.GotFocus += new System.EventHandler(gotFocusEvent);
            this.control.LostFocus += new System.EventHandler(lostFocusEvent);
        }

        public override void initializeNativeControl(System.Xml.XmlElement xmlElement)
        {
            setBounds(Convert.ToInt32(xmlElement.Attributes["xpos"].InnerText),
                      Convert.ToInt32(xmlElement.Attributes["ypos"].InnerText), 
                      Convert.ToInt32(xmlElement.Attributes["width"].InnerText),
                      Convert.ToInt32(xmlElement.Attributes["height"].InnerText));
            setLabel(xmlElement.Attributes["label"].InnerText);

        }


        public void setChecked(bool checkedState)
        {
            
            ((System.Windows.Forms.RadioButton)this.control).Checked = checkedState;
        }

        public bool isChecked()
        {
            return ((System.Windows.Forms.RadioButton)this.control).Checked;
        }
       
        public string getLabel()
        {
            return this.control.Text;
        }

        public void setLabel(string label)
        {
            this.control.Text = label;
        }

    }
}

File: trunk/src/java/YaxglLibrary/de/yaxgl/Controls/RadioButton.java
package de.yaxgl.Controls;

import org.eclipse.swt.SWT;
import org.w3c.dom.Element;

import de.yaxgl.Base.Control;
import de.yaxgl.Container.Container;

public class RadioButton extends Control {

	public RadioButton(Container owner, String ID)
	{
		this.ID=ID;
		this.owner=owner;
		this.control=new org.eclipse.swt.widgets.Button((org.eclipse.swt.widgets.Composite)owner.getNativeComponent(),SWT.RADIO);
		/*register Events*/
		((org.eclipse.swt.widgets.Button)this.control).addSelectionListener(clickEvent(this));
		this.control.addFocusListener(focusEvent(this));
	}
	
	
	@Override
	public void initializeNativeControl(Element xmlElement) {
		setBounds(Integer.valueOf(xmlElement.getAttribute("xpos")),
				Integer.valueOf(xmlElement.getAttribute("ypos")),
				Integer.valueOf(xmlElement.getAttribute("width")),
				Integer.valueOf(xmlElement.getAttribute("height")));

        setLabel(xmlElement.getAttribute("label"));
		
	}
	
	 public void setChecked(boolean checkedState)
     {
		 ((org.eclipse.swt.widgets.Button)this.control).setSelection(checkedState);
     }

     public boolean isChecked()
     {
    	 return ((org.eclipse.swt.widgets.Button)this.control).getSelection();
     }
	
	public String getLabel()
    {
        return ((org.eclipse.swt.widgets.Button)this.control).getText();
    }

    public void setLabel(String label)
    {
   	 ((org.eclipse.swt.widgets.Button)this.control).setText(label);
    }

}
